airbyte-cdk/bulk/toolkits/load-avro/src/main/kotlin/io/airbyte/cdk/load/data/avro/AirbyteValueToAvroRecord.kt

```diff
@@ -27,7 +27,9 @@ class AirbyteValueToAvroRecord {
             is ObjectValue -> {
                 val record = GenericData.Record(schema)
                 airbyteValue.values.forEach { (name, value) ->
-                    record.put(name, convert(value, schema.getField(name).schema()))
+                    schema.getField(name)?.let { field ->
```

Comment:
also what we talked about - avro needs to explicitly filter down to declared fields